### PR TITLE
fix(Scripts/Zul'Aman): Band-Aid fix for hostage loot.

### PR DESCRIPTION
**Pull request by heyitsbench**
## Changes Proposed:

This PR proposes changes to:
-  [X] Scripts (bosses, spell scripts, creature scripts).
-  [X] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/20847

## Tests Performed:

This PR has been:
- [X] Tested in-game by the author.

## How to Test the Changes:

- [X] This pull request can be tested by following the reproduction steps provided in the linked issue.

## Known Issues and TODO List:

- [ ] https://github.com/azerothcore/azerothcore-wotlk/pull/20529

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Kitzunu — APPROVED** [review]

File: src/server/scripts/EasternKingdoms/ZulAman/instance_zulaman.cpp [hunk 1]
```diff
@@ -330,6 +335,7 @@ class instance_zulaman : public InstanceMapScript
         }
 
         private:
+            uint16 ChestLooted;
```
Comment (sudlud) [inline src/server/scripts/EasternKingdoms/ZulAman/instance_zulaman.cpp:338]: should probably be _chestLooted?

Comment (heyitsbench) [reply src/server/scripts/EasternKingdoms/ZulAman/instance_zulaman.cpp]: Corrected.

File: src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp [hunk 2]
```diff
@@ -290,13 +290,16 @@ class npc_zulaman_hostage : public CreatureScript
...
+        creature->GetInstanceScript()->SetData(DATA_CHEST_LOOTED, 0);
+
         float x, y, z;
         creature->GetPosition(x, y, z);
         for (uint8 i = 0; i < 4; ++i)
         {
             if (HostageEntry[i] == creature->GetEntry())
             {
-                creature->SummonGameObject(ChestEntry[i], x - 2, y, z, 0, 0, 0, 0, 0, 0);
+                GameObject* obj = creature->SummonGameObject(ChestEntry[i], x - 2, y, z, 0, 0, 0, 0, 0, 0);
```
Comment (sudlud) [inline src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp:301]: nullptr check

Comment (heyitsbench) [reply src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp:301]: ur a nullptr

Comment (sudlud) [reply src/server/scripts/EasternKingdoms/ZulAman/zulaman.cpp:301]: no u

**Review by sudlud — APPROVED** [review]

**Pull request merged**